### ISSUE TITLE
Add Manual Export feature to export to Maps

### ISSUE THREAD

**Issue by eeverman** [eeverman/andhow#490]
**Is your feature request related to a problem? Please describe.**
Some utilities like [Hibernate can accept configuration as a Properties object](https://docs.jboss.org/hibernate/orm/3.5/javadocs/org/hibernate/cfg/Configuration.html#setProperties(java.util.Properties).  To better support this type of configuration, especially where AndHow is being dropped in to replace an existing custom configuration, it would be nice to export a Group to a Properties or Map object.

**Describe the solution you'd like**
There should be an export utility that:
* Allows one one or more Groups to be specified
* Allows export options (alias or canonical names) to be specified for the entire export and/or individually by Group
* An `@AllowExport` annotation that allows export of all Props in the Group and all subgroups seems like a nice solution.  Its relatively clear in purpose and can be targeted:  If you don't want it to apply to subgroups, just move it to the subgroup you want.  Any current export annotation would be assumed to be allowed as well.  

Further options:
* Nice to support the Stream interface to potentially filter and transform the list (might solve several advanced uses)
* Exports to Java Properties or a Map<String, Object>
* An export to Map<String, String> might be nice too, but might require more advanced configuration to handle String conversion.
* Visibility issues generally go away w/ the addition of the `@AllowExport` annotation.
* Future Feature:  Might be nice to tag aliases for specific exports, but then, its more likely that Properties that are assigned from other Properties will come first, then those can have their own export values.

**Describe alternatives you've considered**
Its possible to mark a Group for Sys.Props export, then grab all Sys.Props and pass them to a 3rd party (i.e. Hibernate).  This is kind of sloppy - Many of those parameters are not related and its possible some 3rd party util will complain about unrecognized params.

Won't help / Won't work:  Another option would be to require this type of export group to extend or implement an interface, like `DirectExport`.  DirectExport could have a static method that iterates through all the fields... But that would still be executed from outside that class, so private members would require privacy breaking reflection.  This also adds an interface requirement where none was before.

**Additional context**
For ref, a GroupExport annotation looks like this:
```
	@GroupExport(
		exporter=SysPropExporter.class,
		exportByCanonicalName=Exporter.EXPORT_CANONICAL_NAME.ONLY_IF_NO_OUT_ALIAS,
		exportByOutAliases=Exporter.EXPORT_OUT_ALIASES.ALWAYS
	)
```
The security aspect needs some thought.  Some options:

1. Only allow Group export for Groups with the export annotation
This implies that there would be some type of no-op exporter implementation, which would just enable exports to be done, but not cause them to happen at startup.  Since a GroupExport / SysPropsExporter annotation already bypasses security by dumping values into Sys.Props, it makes sense to do follow the same pattern and ignore the fact that some Props may be `private`.
2. I do not think there should be a bulk option to export all Groups that have a GroupExport annotation.  Some exports may have been created just for this purpose (export to a Properties) and don't want to be included in a 3rd party libraries bulk map of values.
3.  Is it possible to restrict `private` to exports done within the class that declared them?  Or even better, is it possible to not even have to put an Export annotation on a Group for which you export within it?  For that to work, it would need to match the original metadata (ie which AndHow Properties were declared in which class) to the invocation of export to see if its OK for the caller to invoke it.

**Comment by eeverman** [comment]
This is related to Issue #500

**Comment by eeverman** [comment]
Fixed by #490
